app/flatpak-builtins-remote-delete.c

```diff
@@ -68,6 +68,33 @@ flatpak_builtin_remote_delete (int argc, char **argv, GCancellable *cancellable,
   if (!flatpak_resolve_duplicate_remotes (dirs, remote_name, &preferred_dir, cancellable, error))
     return FALSE;
 
+  if (!opt_force)
+    {
+      g_auto(GStrv) refs = NULL;
+      g_autoptr(GPtrArray) refs_to_remove = NULL;
+      int i;
+
+      refs = flatpak_dir_find_installed_refs (preferred_dir, NULL, NULL, NULL, FLATPAK_KINDS_APP|FLATPAK_KINDS_RUNTIME, 0, NULL);
+      refs_to_remove = g_ptr_array_new_with_free_func (g_free);
+
+      for (i = 0; refs[i]; i++)
```

Comment:
We have to check if `refs` is NULL before indexing into it right? Might as well also pass an error pointer to `flatpak_dir_find_installed_refs`